src/sage/cpython/atexit.pyx

```diff
@@ -154,6 +154,10 @@ cdef extern from *:
     #undef _PyGC_FINALIZED
     #include "internal/pycore_interp.h"
     #include "internal/pycore_pystate.h"
+    #if PY_VERSION_HEX >= 0x030c0000
```

Comment:
I don't have a deep understanding of these things, but reading the comments in `sage_setup/cython_options.py` it appears to me that the usage of PY_VERSION_HEX and PY_MAJOR_VERSION is discouraged. In fact, if I see this correctly than the code added in this PR is the only place now in sage that uses it. @tornaria what's your take on this

Reply:
Uh? This is C code, and `PY_VERSION_HEX` is part of the python stable ABI. See: https://docs.python.org/3/c-api/apiabiversion.html#c.PY_VERSION_HEX.

In fact, I don't think there's any code in sage source that uses `PY_VERSION_HEX` or `PY_MAJOR_VERSION` as defined in `sage_setup/cython_options.py`.

BTW, `git grep PY_VERSION_HEX` shows there are several uses of `PY_VERSION_HEX`, all in C code except for one use in `src/sage/cpython/dict_del_by_value.pyx` which I think it's easy to remove.

Reply:
Thanks a lot. I got thrown off by the usage in `dict_del_by_value` and just did a naive "find all" search. You are right there is nothing wrong with using it in C code.

How would you "fix" `dict_del_by_value`?